app/src/main/java/com/example/androidthings/imageclassifier/ImageClassifierActivity.java

```diff
@@ -73,18 +76,37 @@ protected void onCreate(final Bundle savedInstanceState) {
         mResultViews[1] = (TextView) findViewById(R.id.result2);
         mResultViews[2] = (TextView) findViewById(R.id.result3);
 
+        // Let the user touch the screen to take a photo
```

Comment:
To avoid a weird race condition where the background handler is not initialized for some reason but this click listener is still set, I recommend you to move this setOnClickListener to the end of the init() method.

Reply:
Done.